app/recipe/tests/test_recipe_api.py

```diff
@@ -0,0 +1,168 @@
+"""
+Test for recipe API methods.
+"""
+
+from django.test import TestCase
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from core.models import (Recipe, Ingredient)
+
+RECIPES_URL = reverse('recipe:recipe-list')
+
+
+def detail_url(recipe_id):
+    """Create and return a recipe detail URL."""
+    return reverse('recipe:recipe-detail', args=[recipe_id])
+
+
+def create_recipe(name="Sample Recipe", description="Sample Recipe Description."):
+    """Create and return a sample recipe object."""
+    return Recipe.objects.create(name=name, description=description)
+
+
+def create_ingredient(recipe, name="Sample Ingredient Name"):
+    """Create and return a new ingredient object."""
+    return Ingredient.objects.create(name=name,
+                                     recipe=recipe)
+
+
+class PublicRecipeApiTests(TestCase):
+    """Unauthenticated Recipe API tests"""
```

Comment:
I would add a couple of tests for:

- Try to retrieve an unexistent recipe
- Try to create or update a recipe with wrong data. Example: empty name, or no ingredients.

Reply:
Thanks. Added more tests for the above scenarios.